### DISCUSSION TITLE
How to setup 1 hydra instance with multiple login ui

### DISCUSSION THREAD

**Discussion by thanhpd56** [ory/hydra#2987] [Q&A] [answered]
Hi, 
I have multiple clients (different ui and domain) but share the same hydra.
Each client I want to have different Login UI. 
From idea here (https://github.com/ory/hydra/discussions/2307) we need to create a proxy to redirect to correct login url.
This one can be archieved by read the OAuth2Client info then redirect to correct login url.
However, as I read from model here https://github.com/ory/hydra/blob/dec56ed65ab06372cb97ca94fa24adf69cd9620e/internal/httpclient-next/docs/OAuth2Client.md, there is no property like loginUrl or additional metadata to store the loginUrl.

The question is how can I create each hydra client with loginUrl property?
Please help, thanks.

**Comment by zepatrik** [comment, marked as answer]
If I understand correctly you want to know how the proxy would be able to decide what URI to use. Hydra does not allow storing such information, that is correct. The best way would be to add some kind of persistence to the proxy. When you create the client, you also have to add the login URI to the proxy store.
Alternatively, we could add a kind of metadata raw JSON field to the client in Hydra. But that will be hard with dynamic client registration, as you will have to restrict who can add what metadata. So the first approach seems more reasonable for this use-case.

**Reply by thanhpd56** [reply]
Yes, thank for your help.
